text/0000-default-and-expanded-rustc-errors.md

```diff
@@ -0,0 +1,191 @@
+
+## Tying it together
+
+Lastly, we propose that the final error message:
+
+```
+error: aborting due to 2 previous errors
+```
+
+Be changed to notify users of this ability:
+
+```
+note: You can compile again with --explain for more information about these errors
+```
+
+As this helps inform the user of the --explain capability.
+
+# Drawbacks
+
+Changes in the error format can impact integration with other tools.  For example, IDEs that use a simple regex to detect the error would need to be updated to support the new format.  This takes time and community coordination.
+
+While the new error format has a lot of benefits, it's possible that some errors will feel "shoehorned" into it and, even after careful selection of secondary labels, may still not read as well as the original format.  
+
+There is a fair amount of work involved to update the errors and explain text to the proposed format.  
+
+# Alternatives
+
+Rather than using the proposed error format format, we could only provide the verbose --explain style that is proposed in this RFC.  Famous programmers like [John Carmack](https://twitter.com/ID_AA_Carmack/status/735197548034412546) have praised the Elm error format.
+
+![Image of Elm error](http://www.jonathanturner.org/images/elm_error.jpg)
+
+*Example of an Elm error*
+
+In developing this RFC, we experimented with both styles.  The Elm error format is great as an educational tool, and we wanted to leverage its style in Rust.  For day-to-day work, though, we favor an error format that puts heavy emphasis on quickly guiding the user to what the error is and why it occurred, with an easy way to get the richer explanations (using --explain) when user wants them.
+
+# Stabilization
+
+Currently, these new rust error format is available on nightly using the ```export RUST_NEW_ERROR_FORMAT=true``` environment variable.  Ultimately, this should become the default.  In order to get there, we need to ensure that the new error format is indeed an improvement over the existing format in practice.  
+
+How do we measure the readability of error messages?  This RFC details an educated guess as to what would improve the current state but shows no ways to measure success.
+
+Likewise, While some of us have been dogfooding these errors, we don't know what long-term use feels like.  For example, after a time does the use of color feel excessive?  We can always update the errors as we go, but it'd be helpful to catch it early if possible.
+
+# Unresolved questions
+
+There are a few unresolved questions:
+* Editors that rely on pattern-matching the compiler output will need to be updated.  It's an open question how best to transition to using the new errors.  There is on-going discussion of standardizing the JSON output, which could also be used.
+* Can additional error notes be shown without the "rainbow problem" where too many colors and too much boldness cause errors to beocome less readable?
```

Comment:
I am sure this is a subjective problem. Some people prefer the colorful approach, others not so much, so I think configurability is the key here.